### ISSUE TITLE
Core test should not be dependant in plugins

### ISSUE THREAD

**Issue by purbon** [elastic/logstash#3948]
Core test should not be dependant on plugins to work. This days core test need

```

gem "logstash-filter-clone"
gem "logstash-filter-mutate"
gem "logstash-filter-multiline"
gem "logstash-input-generator"
gem "logstash-input-stdin"
gem "logstash-input-tcp"
gem "logstash-output-stdout"
```
to work, but for this test to be actually validating the pipeline behaviour it should be using mocks or internal dummy filters created for this special purpose. Otherwise when a plugin has a blocker update, test in core are not working as expected.

**Comment by purbon** [comment]
First try to implement it in #3954

**Comment by purbon** [comment]
@suyograo let me know if label is resonable here.

**Comment by suyograo** [comment]
Made it 5.1, since 5.0 after beta1 will be bug fixes only.

**Comment by purbon** [comment]
makes sense, thank for feedback Suyog, also there is another problem here and is we have a ton of Dummy plugins and this internal real plugins, my idea is we should move to something that really make our current test _independant_ of plugins, but we still have integrations test for it. 

Like this we are not blocking anyone when a big refactor happen, or any other small issue in plugins used for test. And also it makes more sense in terms of code separation.

- purbon
